src/main/java/de/tum/in/www1/artemis/service/user/UserCreationService.java

```diff
@@ -91,10 +94,12 @@ public UserCreationService(UserRepository userRepository, PasswordService passwo
      * @param registrationNumber the matriculation number of the student*
      * @param imageUrl           user image url
      * @param langKey            user language
+     * @param isInternal         true if the actual password gets saved in the database
+     * @param initialize         true if the user should be initialized on the first visit
      * @return newly created user
      */
     public User createInternalUser(String login, @Nullable String password, @Nullable Set<String> groups, String firstName, String lastName, String email,
-            String registrationNumber, String imageUrl, String langKey) {
+            String registrationNumber, String imageUrl, String langKey, boolean isInternal, boolean initialize) {
```

Comment:
The method name refers to the fact that it creates something in the internal database. Since it clashes now with the internal flag, I renamed it to `createUser(...)` => we have now two methods, one taking parameters, one a DTO.